tools/awslint/lib/rules/construct.ts

```diff
@@ -227,4 +227,29 @@ constructLinter.add({
       e.assert(!property.name.toLowerCase().endsWith('arn'), `${e.ctx.propsFqn}.${property.name}`);
     }
   }
+});
+
+constructLinter.add({
+  code: 'props-no-tokens',
+  message: 'props should not use the "Token" type',
+  eval: e => {
+    if (!e.ctx.propsType) { return; }
+    if (!e.ctx.hasPropsArgument) { return; }
+
+    // this rule only applies to L2 constructs
+    if (e.ctx.classType.name.startsWith('Cfn')) { return; }
+
+    for (const property of e.ctx.propsType.ownProperties) {
+      const typeRef = property.type;
+      let fqn = typeRef.fqn;
+
+      if (typeRef.arrayOfType) {
+        fqn = typeRef.arrayOfType.fqn;
+      } else if (typeRef.mapOfType) {
+        fqn = typeRef.mapOfType.fqn;
+      }
+
+      e.assert(!(fqn === '@aws-cdk/cdk.Token'), `${e.ctx.propsFqn}.${property.name}`);
```

Comment:
We should probably put all these FQNs into a single file in awslint so if we modify them we have a good place to verify. I would also argue that we should look up this FQN in the typesystem to ensure that it actually exists.